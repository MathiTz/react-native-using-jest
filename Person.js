import React, { Component } from "react";
import { StyleSheet, Text, View } from "react-native";

export default function Person(props) {
  const {
    name,
    age,
    address
  } = props;
  return (
    <View style={styles.container}>
      <Text style={styles.text}>Person Component</Text>
      <Text style={styles.text}>{name}</Text>
      <Text style={styles.text}>{age}</Text>
      <Text style={styles.text}>{address}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#F5FCFF"
  },
  text: {
    fontSize: 20,
    textAlign: "center",
    margin: 10
  }
});
