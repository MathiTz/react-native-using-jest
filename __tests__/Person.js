// __tests__/Person.js
import "react-native";
import React from "react";
import Person from "../Person";

// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Person />).toJSON();
  expect(tree).toMatchSnapshot();
});
