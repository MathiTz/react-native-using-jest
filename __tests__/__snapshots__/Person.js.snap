// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Person Component
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  />
</View>
`;
